blocks/library/image/index.js

```diff
@@ -70,10 +70,9 @@ export const settings = {
 				type: 'raw',
 				isMatch( node ) {
 					const tag = node.nodeName.toLowerCase();
-					const hasText = !! node.textContent.trim();
 					const hasImage = node.querySelector( 'img' );
 
-					return tag === 'img' || ( hasImage && ! hasText ) || ( hasImage && tag === 'figure' );
+					return tag === 'img' || ( hasImage && tag === 'figure' );
```

Comment:
Yes, the same fix is needed for iframes as well as images. Otherwise the image just ends up inside a paragraph. Most of the time, this structure goes together with alignment, so it is not visually clear where the image was anchored.